Bio/PDB/PDBIO.py

```diff
@@ -320,91 +320,95 @@ def save(self, file, select=_select, write_end=True, preserve_atom_numbering=Fal
-                                s = get_atom_line(
-                                    atom,
-                                    hetfield,
-                                    segid,
-                                    atom_number,
-                                    resname,
-                                    resseq,
-                                    icode,
-                                    chain_id,
-                                )
-                            except Exception as err:
-                                # catch and re-raise with more information
-                                raise PDBIOException(
-                                    f"Error when writing atom {atom.full_id}"
-                                ) from err
-                            else:
-                                fhandle.write(s)
-                                # inconsequential if preserve_atom_numbering is True
-                                atom_number += 1
-
-                    if chain_residues_written:
-                        fhandle.write(
-                            _TER_FORMAT_STRING
-                            % (atom_number, resname, chain_id, resseq, icode)
-                        )
-
-                if model_flag and model_residues_written:
-                    fhandle.write("ENDMDL\n")
-            if write_end:
-                fhandle.write("END   \n")
+                        chain_residues_written = 1
+                        model_residues_written = 1
+                        if preserve_atom_numbering:
+                            atom_number = atom.serial_number
+
+                        try:
+                            s = get_atom_line(
+                                atom,
+                                hetfield,
+                                segid,
+                                atom_number,
+                                resname,
+                                resseq,
+                                icode,
+                                chain_id,
+                            )
+                        except Exception as err:
+                            # catch and re-raise with more information
+                            raise PDBIOException(
+                                f"Error when writing atom {atom.full_id}"
+                            ) from err
+                        else:
+                            fhandle.write(s)
+                            # inconsequential if preserve_atom_numbering is True
+                            atom_number += 1
+
+                if chain_residues_written:
+                    fhandle.write(
+                        _TER_FORMAT_STRING
+                        % (atom_number, resname, chain_id, resseq, icode)
+                    )
+
+            if model_flag and model_residues_written:
+                fhandle.write("ENDMDL\n")
+        if write_end:
+            fhandle.write("END   \n")
+
+        if should_close:
```

Comment:
Might be able to use ``isinstance(file, str)`` here instead of a flag variable?

Reply:
Sure, done.